src/main/java/org/jenkinsci/plugins/periodicbackup/S3.java

```diff
@@ -67,16 +69,18 @@
 public class S3 extends Location {
 
     private String bucket;
+    private String prefix;
     private String tmpDir;
     private String region;
     private String credentialsId;
 
     private static final Logger LOGGER = Logger.getLogger(S3.class.getName());
 
     @DataBoundConstructor
-    public S3(String bucket, boolean enabled, String tmpDir, String region, String credentialsId) {
+    public S3(String bucket, String prefix, boolean enabled, String tmpDir, String region, String credentialsId) {
```

Comment:
cannot remember, do we need to create a new constructor to maintain backwards compatibility with old configs..?

Reply:
since it's not a required param, it shouldn't be in the constructor, and you can just create a setter for prefix annotated with @DataBoundSetter